docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,272 @@
+`/path/to/mem_file` contains this time a **diff copy** of the guest memory - the diff
+is consisted of the memory pages which have been dirtied since the last snapshot creation
+or since the creation of the microVM, whichever of these events was the most recent.
+All the other effects mentioned in the **Effects** section from `### Creating full snapshots`
+apply here.
+- _on failure_: no side-effects.
+
+*Note*: To be able to do any incremental (diff) snapshots, you must set `track_dirty_pages`
+field to `true` when configuring the `/machine-config` resource. Example:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i  \
+    -X PUT 'http://localhost/machine-config' \
+    -H 'Accept: application/json'            \
+    -H 'Content-Type: application/json'      \
+    -d '{
+            "vcpu_count": 2,
+            "mem_size_mib": 1024,
+            "ht_enabled": false,
+            "track_dirty_pages": true
+    }'
+```
+
+Enabling this support enables KVM dirty page tracking, so it comes at a cost; you should
+only use it when needed.
+
+Creating a snapshot will **not** influence state, will **not** stop or end the microVM,
+it can be used as before, so you can resume the microVM if you still want to use it.
+
+Resuming the microVM can be done by sending the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PATCH "http://localhost/vm" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+            "state": "Resumed"
+    }"
+```
+
+**Prerequisites**: You can resume the microVM any time after the microVM is booted.
+                   Successive calls of this request keep the microVM in the `Resumed`
+                   state.
+**Effects**:
+- _on success_: microVM is guaranteed to be `Resumed`.
+- _on failure_: no side-effects.
+
+## Loading snapshots
+
+If you want to load a snapshot, you can do that only **before** the microVM is configured
+(the only resource that can be configured prior is the Logger) by sending the following
+API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/load" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+	        \"mem_file_path\": \"/path/to/mem_file\",
+            \"enable_diff_snapshots\": true
+    }"
+```
+
+Details about the required and optional fields can be found in the
+[swagger definition](../../src/api_server/swagger/firecracker.yaml).
+
+**Prerequisites**: Does **not** work with diff snapshots.
+**Effects:**
+- _on success_:
+  - The complete microVM state will be loaded from snapshot into the current Firecracker
+    process.
+  - It resets then the dirtied page bitmap and marks all pages clean (from an incremental
+    snapshot point of view).
+  - The loaded microVM is now in the `Paused` state and needs to be resumed for it to run.
```

Comment:
```suggestion
  - The loaded microVM is now in the `Paused` state.
```

Reply:
I don't know why but it seems to me like an unfinished phrase. I kept it similar with the previous form, but if the second part is too obvious though, I can remove it.